src/plugins/dashboard/public/dashboard_app/no_data/dashboard_app_no_data.tsx

```diff
@@ -44,6 +58,63 @@ export const DashboardAppNoDataPage = ({
       })
     )
   );
+
+  const lensHelpersAsync = useAsync(() => {
+    return lensService?.stateHelperApi() ?? Promise.resolve(null);
+  }, [lensService]);
+
+  const onTryESQL = useCallback(async () => {
+    if (lensHelpersAsync.value) {
+      const { dataViews } = dataService;
+      const indexName = (await getIndexForESQLQuery({ dataViews })) ?? '*';
+      const dataView = await getESQLAdHocDataview(`from ${indexName}`, dataViews);
+      const esqlQuery = getInitialESQLQuery(dataView);
+      const abortController = new AbortController();
+
+      const columns = await getESQLQueryColumns({
```

Comment:
What happens if getESQLQueryColumns throws? I think we should handle the error case as well.

Reply:
Done https://github.com/elastic/kibana/pull/196658/commits/f806040487bcc820e8680b3e4c4af369ae5fe0c0